### PR TITLE
add modulo and power operators

### PR DESCRIPTION
**Pull request by gserena01**
related to https://github.com/KittyCAD/modeling-app/pull/1341

### CODE REVIEW

File: execution-plan/src/arithmetic.rs [hunk 1]
```diff
@@ -55,6 +55,30 @@ impl UnaryArithmetic {
...
+
+macro_rules! power_int_impl {
+    ($($t:ty)*) => ($(
+        impl Power for $t {
+            fn power(self, other: $t) -> $t { self.overflowing_pow(other as u32).0 }
+        }
+
+    )*)
+}
+power_int_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
```
Comment (adamchalmers) [inline execution-plan/src/arithmetic.rs:80]: Could you please add some unit tests to check the behaviour with negative numbers?

File: execution-plan/src/tests.rs [hunk 2]
```diff
@@ -191,6 +191,142 @@ async fn add_to_composite_value() {
...
+                operand0: Operand::Reference(Address::ZERO),
+                operand1: Operand::Literal((-4.2f32).into()),
+            },
+            destination: Destination::Address(Address::ZERO + 1),
+        },
+    ];
+    // (-2.5)^-4.2 = NaN
+    let mut mem = Memory::default();
+    execute(&mut mem, plan, None).await.expect("failed to execute plan");
+    // TODO: This test is failing because "  left: Some(NaN) right: Some(NaN)"
```
Comment (gserena01) [inline execution-plan/src/tests.rs:282]: Why might this test be failing? I've tried expecting f32::NAN and f64::NAN, but neither is equivalent to the actual NAN

Comment (adamchalmers) [reply execution-plan/src/tests.rs:282]: NaN is always != to other NaN I believe (@paultag might know more). You could use the [`is_nan()` method](https://doc.rust-lang.org/stable/std/primitive.f64.html#method.is_nan)

Comment (paultag) [reply execution-plan/src/tests.rs:282]: Yes! From the floating point IEEE spec -- any comparison with NaN is false on either side, even `NaN == NaN`, or `NaN < Inf`, etc.

Screenshot from 2024-02-05 21-08-42

File: execution-plan/src/tests.rs [hunk 3]
```diff
@@ -191,6 +191,142 @@ async fn add_to_composite_value() {
...
+                operand0: Operand::Reference(Address::ZERO),
+                operand1: Operand::Literal((-4.2f32).into()),
+            },
+            destination: Destination::Address(Address::ZERO + 1),
+        },
+    ];
+    // (-2.5)^-4.2 = NaN
+    let mut mem = Memory::default();
+    execute(&mut mem, plan, None).await.expect("failed to execute plan");
+    // let result = f32::try_from(mem.get(&(Address::ZERO + 1)).unwrap());
```
Comment (gserena01) [inline execution-plan/src/tests.rs:282]: Okay! So here I'm trying to use is_nan(), but I'm getting the types wrong. I'm seeing an error that try_from is not implemented for f32, but it appears to have been implemented in primitive.rs in the modeling-app(?) repository. Am I missing something else here?

**Review by adamchalmers — APPROVED** [review]
